### ISSUE TITLE
how do you decide loss weight?

### ISSUE THREAD

**Issue by timemao** [wangruixing/DeepUPE#31]
Hi, you did good work!

But I have trouble with loss weight, the L2-loss(reconstruction loss) finally converge to about 1000  for 16*512*512*3 [0,1],but the TV-loss calculated from groundtruth illumination map (get by input/gt image) and input is about 10^6, so we cannot place 1 for L2-loss and 2 for Tv-loss.

Any advise?

**Comment by wangruixing** [comment]
Did you mean L2 loss is around 1000? tv loss is around 10^6?  In tv loss(local smooth loss) function, the input is origin image, the output is the illumination map. Actually, during my training process, the loss is around 0.033. I suggest you can only use the input image pixel divide illumination map at first, then calculate the l2 loss between the predicted image and ground truth. Only use l2 loss and make sure the loss is right at first.

**Comment by timemao** [comment]
Could you make sure metrics.py tv-loss code?

**Comment by timemao** [comment]
![4](https://user-images.githubusercontent.com/23051743/61629752-63cf4a00-acb8-11e9-8034-6e04945429b6.jpg)
![4_gt](https://user-images.githubusercontent.com/23051743/61629753-63cf4a00-acb8-11e9-9fed-87e3955732a8.jpg)
I calculate these two images tv-loss in MATLAB，my code and result:
![image](https://user-images.githubusercontent.com/23051743/61629949-e5bf7300-acb8-11e9-943a-34ba93ddae9b.png)

**Comment by wangruixing** [comment]
It is the local smoothness code, not standard tv loss.

**Comment by wangruixing** [comment]
The second parameter is not a predicted image, it is the illumination map.

**Comment by timemao** [comment]
Actually I calculate illumination map with
_______________________________________________
S=min(1.0,max(im1d,im1d./(im2d+0.0001)));
_______________________________________________
You mean the second parameter not like this? In metrics.py, you use tf.image.rbg_to_grayscale, this function get 1-channal output

**Comment by wangruixing** [comment]
You forgot to calculate the mean. tf.reduce_mean. and during processing image, the scale is 0-1.
